### PR TITLE
Change Maven hostname to maven.pkg.github.com

### PR DESCRIPTION
**Pull request by jcansdale**
### Why:

Make Maven hostname consistent with other ecosystems (`maven.pkg.github.com`).

As suggested here https://github.zendesk.com/agent/tickets/851339

### What's being changed:

- Change Maven host to `maven.pkg.github.com`.
- Change `rubygem.pkg.github.com` to `rubygems.pkg.github.com`.
- Add missing ` to `nuget.pkg.github.com`.

### Check off the following:
- [ ] All of the tests are passing.
- [ ] I have reviewed my changes in staging.
- [ ] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [ ] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
👋 Thanks for opening a PR @jcansdale! Would you mind resolving the conflicts? I'll get this triaged for review ⚡

**Comment by jcansdale** [comment]
@janiceilene done. 😄  I don't know why that was conflicted!

File: content/packages/publishing-and-managing-packages/about-github-packages.md [hunk 1]
```diff
@@ -103,8 +103,8 @@ If {% data variables.product.product_location %} has subdomain isolation disable
 | --- | --- | --- | --- | --- |
 | JavaScript | Node package manager | `package.json`  | `npm` | `npm.pkg.github.com/OWNER/REPOSITORY/IMAGE-NAME`
 | Ruby |  RubyGems package manager | `Gemfile` |  `gem` | `rubygems.pkg.github.com/OWNER/REPOSITORY/IMAGE-NAME`
-| Java | Apache Maven project management and comprehension tool | `pom.xml` |  `mvn` | `maven.HOSTNAME/OWNER/REPOSITORY/IMAGE-NAME`
-| Java | Gradle build automation tool for Java | `build.gradle` or `build.gradle.kts`  | `gradle`  | `maven.HOSTNAME/OWNER/REPOSITORY/IMAGE-NAME`
```
Comment (lucascosti) [inline content/packages/publishing-and-managing-packages/about-github-packages.md:107]: 😬 eeep, looks like this probably crept in from the GHES versioning...

**Review by lucascosti — APPROVED** [review]
Thanks a lot for these fixes, @jcansdale. I'll get them merged in 🚀

**Pull request merged**